crates/block-producer/src/block_producer.rs

```diff
@@ -464,17 +556,18 @@ impl BlockProducer {
             .push((generated_stake.output, generated_stake.output_data));
 
         // withdrawal cells
-        let generated_withdrawal_cells = crate::withdrawal::generate(
-            &rollup_cell,
+        let mut generated_withdrawal_cells = crate::withdrawal::generate(
             rollup_context,
             &block,
             &self.config,
-            &self.rpc_client,
-        )
-        .await?;
-        tx_skeleton
-            .cell_deps_mut()
-            .extend(generated_withdrawal_cells.deps);
+            collected_custodians,
+        )?;
+        let mut resolved_deps =
+            resolve_type_deps(&self.rpc_client, &generated_withdrawal_cells.inputs).await?;
```

Comment:
Which type_deps do you expect to resolve? IMO there are only two deps: withdraw-lock & sUDT-type. We can pass these two from configure file.

Reply:
This would be a breaking change to ```config.toml```. I think we can add ```ckb-indexer-url``` to ```gw-tools``` to generate cell dep for user.

Reply:
Done. Now user should set ```l1_sudt_tyep_dep``` in ```config.toml```.